### PR TITLE
FIPS-8557/FIPS-8887 :: split abstraction

### PR DESCRIPTION
**Pull request by hd42**
Change-Id: Ibdcacf84980b9dfdeb77bcec5f36f7cc83f5cdfa

### CODE REVIEW

File: org.faktorips.aggregator/tests/pom.xml [hunk 1]
```diff
@@ -71,6 +71,11 @@
             <artifactId>faktorips-fl</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (laeubi) [reply org.faktorips.aggregator/tests/pom.xml:74]: Ich sehe im log output:

> Unresolved requirement: Require-Bundle: org.eclipse.osgi; bundle-version="[3.17.0,4.0.0)"; visibility:="reexport"

prüfe mal im manifest das du einen import zu `org.osgi.framework` hast und kein require bundle zu `eclipse.osgi` das scheint mir doch etwas merkwürding.

[the remark this answers is not in the document]
Comment (hd42) [reply org.faktorips.aggregator/tests/pom.xml:74]: Klappt leider nur, wenn ich das zusätzlich in die Targetplatform packe. Dann gibt's aber den nächsten Fehler:
`java.lang.NoClassDefFoundError: org/eclipse/core/runtime/IStatus
	at org.faktorips.devtools.abstraction.Abstractions.lambda$2(Abstractions.java:47)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at org.faktorips.devtools.abstraction.Abstractions.get(Abstractions.java:47)
	at org.faktorips.devtools.abstraction.Abstractions.<clinit>(Abstractions.java:28)
	at org.faktorips.abstracttest.PlatformProjectBuilder.newPlatformProject(PlatformProjectBuilder.java:57)
	at org.faktorips.abstracttest.PlatformProjectBuilder.build(PlatformProjectBuilder.java:49)
	at org.faktorips.devtools.abstraction.eclipse.internal.EclipseAbstractionTestSetup.createEclipseProject(EclipseAbstractionTestSetup.java:81)
	at org.faktorips.devtools.abstraction.eclipse.internal.EclipseAbstractionTestSetup.newProjectImpl(EclipseAbstractionTestSetup.java:45)
	at org.faktorips.devtools.abstraction.testsetup.AbstractAbstractionTestSetup.newAbstractionProject(AbstractAbstractionTestSetup.java:150)
	at org.faktorips.devtools.abstraction.eclipse.internal.EclipseContainerTest.setUp(EclipseContainerTest.java:36)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.pde.internal.junit.runtime.RemotePluginTestRunner.main(RemotePluginTestRunner.java:232)
	at org.eclipse.pde.internal.junit.runtime.PlatformUITestHarness.lambda$0(PlatformUITestHarness.java:45)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4582)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:566)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.runApp(NonUIThreadTestApplication.java:55)
	at org.eclipse.pde.internal.junit.runtime.UITestApplication.runApp(UITestApplication.java:46)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.start(NonUIThreadTestApplication.java:49)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:661)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:597)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1476)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1449)
Caused by: java.lang.ClassNotFoundException: org.eclipse.core.runtime.IStatus cannot be found by org.faktorips.devtools.abstraction_22.6.0.SNAPSHOT
	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:511)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:422)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:153)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)`

org.eclipse.core ist aber schon im target und als Dependency eingetragen.

[the remark this answers is not in the document]
Comment (laeubi) [reply org.faktorips.aggregator/tests/pom.xml:74]: das abstraction sollte doch ein plain maven projekt sein? Das sollte unahängig von der TP sein.

[the remark this answers is not in the document]
Comment (hd42) [reply org.faktorips.aggregator/tests/pom.xml:74]: > das abstraction sollte doch ein plain maven projekt sein? Das sollte unahängig von der TP sein.

Ja, aber beim Ausführen/Testen der darauf basierenden Eclipse-Plugins org.faktorips.devtools.abstraction.eclipse bzw. org.faktorips.devtools.model(.test) gibt es Probleme, dass diese Code aus dem abstraction-Projekt aufrufen welches dann scheinbar Klassen nicht laden kann, die es als Maven-Dependencies hat und die auch in der Target-Platform enthalten sind.

**Pull request merged**